examples/step-22/step-22.cc

```diff
@@ -729,22 +736,17 @@ namespace Step22
                 // is overloaded for symmetric tensors, yielding the scalar
                 // product between the two tensors.
                 //
-                // For the right-hand side we use the fact that the shape
-                // functions are only non-zero in one component (because our
-                // elements are primitive).  Instead of multiplying the tensor
-                // representing the dim+1 values of shape function i with the
-                // whole right-hand side vector, we only look at the only
-                // non-zero component. The function
-                // FiniteElement::system_to_component_index will return
-                // which component this shape function lives in (0=x velocity,
-                // 1=y velocity, 2=pressure in 2d), which we use to pick out
-                // the correct component of the right-hand side vector to
-                // multiply with.
-                const unsigned int component_i =
-                  fe.system_to_component_index(i).first;
-                local_rhs(i) += (fe_values.shape_value(i, q)   // (phi_u_i(x_q)
-                                 * rhs_values[q](component_i)) // * f(x_q))
-                                * fe_values.JxW(q);            // * dx
+                //
+
+                // For the right-hand side, even if our elements are primitives,
+                // we multiply the tensor representing the first dim values of
+                // shape function i (the ones corresponding to the velocity)
+                // with the first dim components of the right-hand side vector,
+                // both evaluated at the current quadrature point.
+
```

Comment:
Actually, whether or not we use primitive elements here no longer matters when you use the extractors. Maybe something like this:
```suggestion
                // For the right-hand side, we need to multiply the (vector of) velocity shape functions
                // with the vector of body force right-hand side components,
                // both evaluated at the current quadrature point.
                // We have implemented the body forces as a `TensorFunction<1,dim>`, so its
                // values at quadrature points are already tensors for which the application of 
                // `operator*` against the velocity components of the shape function results in
                // the dot product, as intended.
```